artemis-jms/src/main/java/org/acme/artemis/PriceConsumer.java

```diff
@@ -0,0 +1,57 @@
+package org.acme.artemis;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSConsumer;
+import javax.jms.JMSContext;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.Session;
+
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+/**
+ * A bean consuming prices from the JMS queue.
+ */
+@ApplicationScoped
+public class PriceConsumer implements Runnable {
+
+    @Inject
+    ConnectionFactory connectionFactory;
+
+    private final ExecutorService scheduler = Executors.newSingleThreadExecutor();
+
+    private volatile String lastPrice;
+
+    public String getLastPrice() {
+        return lastPrice;
+    }
+
+    void onStart(@Observes StartupEvent ev) {
+        scheduler.submit(this);
+    }
+
+    void onStop(@Observes ShutdownEvent ev) {
+        scheduler.shutdown();
+    }
+
+    @Override
+    public void run() {
+        try (JMSContext context = connectionFactory.createContext(Session.AUTO_ACKNOWLEDGE)) {
+            JMSConsumer consumer = context.createConsumer(context.createQueue("prices"));
+            while (true) {
+                Message message = consumer.receive();
+                if (message == null) return;
```

Comment:
Can you add a comment explaining that `message==null` is a termination signal?

Reply:
Yes, good idea.